packages/performance/index.ts

```diff
@@ -76,9 +83,28 @@ declare module '@firebase/app-types' {
   interface FirebaseNamespace {
     performance?: {
       (app?: FirebaseApp): FirebasePerformance;
+      isSupprted(): Promise<boolean>;
     };
   }
   interface FirebaseApp {
     performance?(): FirebasePerformance;
   }
 }
+
+async function isSupported(): Promise<boolean> {
```

Comment:
Performance is a monitoring sdk. It is easier for the users to initialize it in all environments, but performance will not start internally if all the apis it needs are not available.
Because of that, we probably do not need the isSupported function so the user can check before initializing.

Reply:
Analytics is in a similar situation but it throws if the environment conditions aren't met, on the assumption the developer might want to be alerted if they're not getting analytics data from a certain type of environment. `isSupported()` is a way to prevent initialization and avoid users having to see those errors. Maybe analytics could make some changes in the future?

But for now, since Performance is designed to surface this through `console.info` which won't appear for most users, I guess we don't need a mechanism to suppress it. Also performance internal initialization can be blocked by an async check (unlike analytics) so it seems like all the checks and blocking can be handled without developers or users seeing anything.

Reply:
I removed the `isSupported` function